driver/bpf/probe.c

```diff
@@ -36,48 +36,90 @@ __bpf_section(TP_NAME prefix #event)			\
 int bpf_##event(struct type *ctx)
 #endif
 
+#define __NR_ia32_socketcall 102
+
 BPF_PROBE("raw_syscalls/", sys_enter, sys_enter_args)
 {
-	const struct syscall_evt_pair *sc_evt;
-	ppm_event_code evt_type;
-	int drop_flags;
-	long id;
-	bool enabled;
-
-	if (bpf_in_ia32_syscall())
-		return 0;
+	const struct syscall_evt_pair *sc_evt = NULL;
+	ppm_event_code evt_type = -1;
+	int drop_flags = 0;
+	long id = 0;
+	bool enabled = false;
+	int socketcall_syscall_id = -1;
+
+#ifdef __NR_socketcall
+	socketcall_syscall_id = __NR_socketcall;
+#endif
 
 	id = bpf_syscall_get_nr(ctx);
 	if (id < 0 || id >= SYSCALL_TABLE_SIZE)
 		return 0;
 
-#if defined(CAPTURE_SOCKETCALL) && defined(BPF_SUPPORTS_RAW_TRACEPOINTS)
-	if(id == __NR_socketcall)
+	if (bpf_in_ia32_syscall())
 	{
-		id = convert_network_syscalls(ctx);
-	}
+		if (id == __NR_ia32_socketcall)
+		{
+			socketcall_syscall_id = __NR_ia32_socketcall;
+		}
+		else
+		{
+#ifdef CONFIG_X86_64
+			id = convert_ia32_to_64(id);
+			if(id == -1)
+			{
+				return 0;
+			}
+#else
+			// TODO: unsupported
+			return 0;
 #endif
+		}
+	}
+
+	if(id == socketcall_syscall_id)
+	{
+		bool is_syscall_return = false;
+		int return_code = convert_network_syscalls(ctx, &is_syscall_return);
+		/* If we return an event code, it means we need to call directly `record_event_all_consumers` */
+		if(!is_syscall_return)
+		{
+			evt_type = return_code;
+			drop_flags = return_code == PPME_GENERIC_E ? UF_ALWAYS_DROP : UF_USED;
+		}
+		else
+		{
+			id = return_code;
+		}
+	}
 
 	enabled = is_syscall_interesting(id);
```

Comment:
we need to handle the case in which `id==__NR_ia32_socketcall`

Reply:
Fixed in 6e1513b3b7d74a8b6d60bf51d4a76ef49ba3479c